packages/url/src/is-url.js

```diff
@@ -10,8 +8,18 @@ const URL_REGEXP = /^(?:https?:)?\/\/\S+$/i;
  * const isURL = isURL( 'https://wordpress.org' ); // true
  * ```
  *
+ * @see https://url.spec.whatwg.org/
+ * @see https://url.spec.whatwg.org/#valid-url-string
+ *
  * @return {boolean} Whether or not it looks like a URL.
  */
 export function isURL( url ) {
-	return URL_REGEXP.test( url );
+	// A URL can be considered value if the `URL` constructor is able to parse
+	// it. The constructor throws an error for an invalid URL.
+	try {
+		new URL( url );
+		return true;
+	} catch ( error ) {
```

Comment:
Note: Originally, I wanted to implement this as (the perfectly-valid):

```suggestion
	} catch {
```

However, this is blocked by #19952 . We can choose to circle back and refactor this after that task is resolved.